micrometer-tracing-tests/micrometer-tracing-test/src/test/java/io/micrometer/tracing/test/simple/SpanAssertTests.java

```diff
@@ -488,6 +488,39 @@ void should_fail_when_spanId_is_not_equal() {
         thenThrownBy(() -> assertThat(span).hasSpanIdEqualTo("2")).isInstanceOf(AssertionError.class);
     }
 
+    @Test
```

Comment:
Could you please add a few more tests both for `hasParentIdEqualTo` and `doesNotHaveParentIdEqualTo` to cover all the possible cases?

| Actual  | Expected |
| ------------- | ------------- |
| null  | null  |
| null  | non-null  |
| non-null  | null  |
| non-null  | non-null (they are equal)  |
| non-null  | non-null (they are not equal)  |

Reply:
Ok, think I've covered all the cases now!